draft-pauly-httpbis-geoip-hint.md

```diff
@@ -73,63 +95,99 @@ For example, the header for an entry "192.0.2.5,US,US-AL,Alabaster" would be:
-{{?RFC7871}}.
+If possible, the client SHOULD specify their geolocation. If location
+is not available, the client MAY send a default value or none at all.
+How the default value is determined is outside the scope of this
+document. However, the default value MUST NOT be more precise or
+detailed than what could be inferred from the user’s IP address.
 
-# Security Considerations {#security}
+The client MAY include the client hint in requests to the server after
+the server has explicitly opted in to receiving the hint, or if the
+client knows of specific server configurations, such as proxy
+settings, that support including the hint.
 
-The use of the IP Geolocation Client Hint MUST use the Sec- header prefix as recommended
-in {{!RFC8942}}.
+# Server Behavior
 
-Servers MUST NOT use IP Geolocation Client Hints for making security or access-control decisions,
-as the value can be spoofed by a client. The hint is intended only for use in optimizing behavior.
+Upon receiving a Geolocation Client Hint, a server can use the
+information to influence its behavior in various ways, such as
+determining the content of HTTP responses.
+
+Servers can choose to use the hint value in one of several ways,
+including:
+
+- Using the client hint information instead of consulting IP-based
+geolocation feeds.
+- Serving content that corresponds to the client’s indicated location,
+including delivering region-specific news, weather forecasts, and
+relevant advertisements.
+- Determining service availability and feature access based on the
+client’s indicated location.
+
+The server MUST be able to handle situations where geolocation is
+not provided in a request. Since not all web clients, such as curl,
+will send a Geolocation Client, the server MAY defer to alternative
+methods such as IP-based geolocation feeds to provide said value.
+
+If the server is acting as a forward proxy, such as a CONNECT proxy,
+it can use the hint to determine an appropriate geo-mapped IP address
+to use for outbound connections, or a client subnet to present in the
+EDNS0 Client Subnet extension for DNS queries {{?RFC6891}}
+{{?RFC7871}}.
 
-The value contained in this hint SHOULD be based only on a IP Geolocation feed value for
-an IP address the client is already presenting to a server. In order to avoid disclosing
-any private information, this value MUST not be based on geolocation of the client determined
-by other means, such as physical latitude and longitude coordinates.
+# Security Considerations {#sec-considerations}
+
+The use of the Geolocation Client Hint MUST use the `Sec-` header
+prefix as recommended in {{!RFC8942}}.
+
+Servers MUST NOT use Geolocation Client Hints for making security or
+access-control decisions, as the value is provided by the client with
+no additional authentication. The hint is intended only to be used
+for greater user visibility and say over their geolocation.
```

Comment:
I think we should stay away from user visibility and say. The intent here isn't to allow the human user to control this